HPX does not compile on the new Mac OSX using the M1 chip

## Expected Behavior

HPX should compile on ARM, since it works on Fedora and the A64FX machine, I have access too. 

## Actual Behavior

However, on the new Apple M1 chip, it does not might be due to a different name of the hardware type. 

```
[ 13%] Building CXX object libs/core/threading_base/CMakeFiles/hpx_threading_base.dir/src/thread_pool_base.cpp.o
/Users/diehlpk/git/hpx/libs/core/threading_base/src/thread_pool_base.cpp:96:46: error: no member named 'hardware' in namespace 'hpx::util'
        std::uint64_t base_timestamp = util::hardware::timestamp();
                                       ~~~~~~^
/Users/diehlpk/git/hpx/libs/core/threading_base/src/thread_pool_base.cpp:98:46: error: no member named 'hardware' in namespace 'hpx::util'
        std::uint64_t curr_timestamp = util::hardware::timestamp();
                                       ~~~~~~^
/Users/diehlpk/git/hpx/libs/core/threading_base/src/thread_pool_base.cpp:103:36: error: no member named 'hardware' in namespace 'hpx::util'
            curr_timestamp = util::hardware::timestamp();
                             ~~~~~~^
3 errors generated.
make[2]: *** [libs/core/threading_base/CMakeFiles/hpx_threading_base.dir/src/thread_pool_base.cpp.o] Error 1
make[1]: *** [libs/core/threading_base/CMakeFiles/hpx_threading_base.dir/all] Error 2
make: *** [all] Error 2
```



## Steps to Reproduce the Problem

  1. Install all dependencies using homebrew
  1. Run cmake


## Specifications

  - HPX Version: 1.6 or master
  - Platform (compiler, OS): Mac OSX Big Sur and M1 chip

--- a/libs/core/hardware/include/hpx/hardware/timestamp/linux_generic.hpp
+++ b/libs/core/hardware/include/hpx/hardware/timestamp/linux_generic.hpp
@@ -8,8 +8,6 @@
 
 #pragma once
 
-#if defined(__unix__)
-
 #include <cstdint>
 
 #include <time.h>
@@ -24,5 +22,3 @@ namespace hpx { namespace util { namespace hardware {
     }
 
 }}}    // namespace hpx::util::hardware
-
-#endif

